addons/mail/static/src/composer/suggestion_service.js

```diff
@@ -142,7 +142,48 @@ export class SuggestionService {
                 }
             }
         }
-        const sortFunc = (p1, p2) => {
+        return {
+            type: "Partner",
+            mainSuggestions: sort
+                ? this.sortPartnerSuggestions(mainSuggestionList, cleanedSearchTerm, thread)
+                : mainSuggestionList,
+            extraSuggestions: sort
+                ? this.sortPartnerSuggestions(extraSuggestionList, cleanedSearchTerm, thread)
+                : extraSuggestionList,
+        };
+    }
+
+    /**
+     * @param {[import("@mail/core/persona_model").Persona]} [partners]
+     * @param {String} [searchTerm]
+     * @param {import("@mail/core/thread_model").Thread} thread
+     * @returns {[import("@mail/core/persona_model").Persona]}
+     */
+    sortPartnerSuggestions(partners, searchTerm = "", thread = undefined) {
+        const cleanedSearchTerm = cleanTerm(searchTerm);
+        return partners.sort((p1, p2) => {
+            const mostResentChatPartners = this.personaService.getMostRecentChatPartnerIds();
+            const positionAInMostRecentChatPartners = mostResentChatPartners.findIndex(
+                (partnerId) => partnerId === p1.id
+            );
+            const positionBInMostRecentChatPartners = mostResentChatPartners.findIndex(
+                (partnerId) => partnerId === p2.id
+            );
+            if (
+                positionAInMostRecentChatPartners !== -1 &&
+                positionBInMostRecentChatPartners === -1
+            ) {
+                return -1;
+            } else if (
+                positionAInMostRecentChatPartners === -1 &&
+                positionBInMostRecentChatPartners !== -1
+            ) {
+                return 1;
+            } else if (positionAInMostRecentChatPartners < positionBInMostRecentChatPartners) {
+                return -1;
+            } else if (positionAInMostRecentChatPartners > positionBInMostRecentChatPartners) {
+                return 1;
+            }
```

Comment:
variable names are too long, which makes code hard to read.
`recentChatIndex_p1` / `recentChatIndex_p2` is good enough and makes it easier to read.
Also less LOCs from prettier formatting